src/Cnab.Bundle/Parameter.cs

```diff
@@ -27,42 +50,82 @@ public class ParameterMetadata
         public string Description { get; set; }
     }
 
-    public class StringParameter : Parameter
+    public class StringParameter : Parameter<string>
     {
-        [JsonProperty("defaultValue")]
-        public string DefaultValue { get; set; }
-
-        [JsonProperty("allowedValues")]
-        public List<string> AllowedValues { get; set; }
-
         [JsonProperty("minLength")]
         public int MinimumLength { get; set; }
 
         [JsonProperty("maxLength")]
         public int MaximumLength { get; set; }
-    }
 
-    public class BoolParameter : Parameter
-    {
-        [JsonProperty("defaultValue")]
-        public bool DefaultValue { get; set; }
+        public override bool IsValid(string value)
+        {
+            if (!this.AllowedValues.Contains(value))
+                return false;
 
-        [JsonProperty("allowedValues")]
-        public List<bool> AllowedValues { get; set; }
+            if ((value.Length < this.MinimumLength) || (value.Length > this.MaximumLength))
+                return false;
+
+            return true;
+        }
+
+        public override bool IsValid(bool value)
+        {
+            return false;
+        }
+
+        public override bool IsValid(int value)
+        {
+            return false;
+        }
     }
 
-    public class IntParameter : Parameter
+    public class BoolParameter : Parameter<bool>
     {
-        [JsonProperty("defaultValue")]
-        public int DefaultValue { get; set; }
+        public override bool IsValid(bool value)
+        {
+            return true;
+        }
 
-        [JsonProperty("allowedValues")]
-        public List<int> AllowedValues { get; set; }
+        public override bool IsValid(string value)
+        {
+            return false;
+        }
 
+        public override bool IsValid(int value)
+        {
+            return false;
+        }
+
+    }
+
+    public class IntParameter : Parameter<int>
+    {
         [JsonProperty("minValue")]
         public int MinimumValue { get; set; }
 
         [JsonProperty("maxValue")]
         public int MaximumValue { get; set; }
+
+        public override bool IsValid(int value)
+        {
+            if (!this.AllowedValues.Contains(value))
+                return false;
+
+            if ((value < this.MinimumValue) || (value > this.MaximumValue))
+                return false;
+
+            return true;
+        }
+
+        public override bool IsValid(string value)
+        {
+            return false;
```

Comment:
Right now, "80" is an invalid value for an integer parameter. Should we try parsing it as an integer first?